File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/atoms/placeholder/roles/TextPlaceholderRole.kt
package com.perrystreet.woof.designsystem.atomic.atoms.placeholder.roles

import androidx.compose.runtime.Composable
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.Dp
import com.perrystreet.woof.designsystem.theme.Theme

enum class TextPlaceholderRole(val sampleText: String) {
    Title(sampleText = "Placeholder title") {
        @Composable
        override fun textStyle(): TextStyle = Theme.typography.display.h4

        @Composable
        override fun radius(): Dp = Theme.radius.s
    },
    Subtitle(sampleText = "Placeholder subtitle text") {
        @Composable
        override fun textStyle(): TextStyle = Theme.typography.subhead.p2

        @Composable
        override fun radius(): Dp = Theme.radius.s
    },
    Body(sampleText = "Placeholder body text that spans a line") {
        @Composable
        override fun textStyle(): TextStyle = Theme.typography.body.p1

        @Composable
        override fun radius(): Dp = Theme.radius.s
    },
    ;

    @Composable
    abstract fun textStyle(): TextStyle

    @Composable
    abstract fun radius(): Dp
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Atoms/Placeholder/Roles/TextPlaceholderRole.swift
import SwiftUI

public enum TextPlaceholderRole {
    case title
    case subtitle
    case body

    public var sampleText: String {
        switch self {
        case .title: "Placeholder title"
        case .subtitle: "Placeholder subtitle text"
        case .body: "Placeholder body text that spans a line"
        }
    }

    public func font(from theme: ThemeImplementing) -> Font {
        switch self {
        case .title: theme.typography.display.h4
        case .subtitle: theme.typography.subhead.p2
        case .body: theme.typography.body.p1
        }
    }
}
